lookout/core/metrics/values.py

```diff
@@ -0,0 +1,129 @@
+import os
+from threading import Lock
+
+from prometheus_client.mmap_dict import mmap_key, MmapedDict
+
+"""This file is a copy of
+  https://github.com/prometheus/client_python/blob/master/prometheus_client/values.py
+  modified to increment the variables using the Kahan sum algorithm to maintain precision even
+  when aggregating long sequences of numbers."""
+
+
+class MutexValue(object):
+    """A float protected by a mutex."""
+
+    _multiprocess = False
+
+    def __init__(self, typ, metric_name, name, labelnames, labelvalues, **kwargs):
+        self._value = 0.0
+        self._lock = Lock()
+        self.__interm_sum = 0
+
+    def inc(self, amount):
+        with self._lock:
+            summ = self._value
+            y = amount - self.__interm_sum
+            t = summ + y
+            self.__interm_sum = (t - summ) - y
+            self._value = t
+
+    def set(self, value):
+        with self._lock:
+            self.__interm_sum = 0
+            self._value = value
+
+    def get(self):
+        with self._lock:
+            return self._value
+
+
+def MultiProcessValue(_pidFunc=os.getpid):
+    files = {}
+    values = []
+    pid = {"value": _pidFunc()}
+    # Use a single global lock when in multi-processing mode
+    # as we presume this means there is no threading going on.
+    # This avoids the need to also have mutexes in __MmapDict.
+    lock = Lock()
```

Comment:
Oh no this is no good, the lock should always be individual. Otherwise concurrent submitting different metrics will be slower.